File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows ==1:
            return [[1]]
        result=[[1],[1,1]]
        list1=[]
        sums=0
        for i in range(2,numRows):
            list1=[]
            list1.append(1)
            for j in range(i-1):
                sums=sums+sum(result[-1][j:j+2])
                list1.append(sums)
                sums=0
            list1.append(1)
            result.append(list1)
        return result
        

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> arr=new ArrayList<>();
        arr.add(new ArrayList<>(Arrays.asList(1)));
        if (numRows==1)
        {
            return arr;
        }
        arr.add(new ArrayList<>(Arrays.asList(1,1)));
        System.out.println(arr);
        int sum=0;
        for(int i=2;i<numRows;i++)
        {
            ArrayList<Integer>arr1=new ArrayList<>();
            arr1.add(1);
            for(int j=0;j<i-1;j++)
            {
                sum=0;
                List<Integer>a=arr.get(i-1);
                sum=sum+a.get(j)+a.get(j+1);
                arr1.add(sum);
            }
            arr1.add(1);
            arr.add(arr1);
        }
        return arr;
    }
}